doc/rst/technotes/gpu.rst

```diff
@@ -136,39 +139,125 @@ GPU-Related Environment Variables
-If the type of GPU is not detected you may set ``CHPL_GPU`` manually to either
-``nvidia``, ``amd`` or ``cpu`` (See :ref:`below <cpu-mode-label>` for more
-information on this mode). If the relevant runtime path is not automatically
-detected (or you would like to use a different installation) you may set
-``CHPL_CUDA_PATH`` and/or ``CHPL_ROCM_PATH``.
-
-``CHPL_GPU_ARCH`` environment variable can be set to control the desired GPU
-architecture to compile for. For ``CHPL_GPU=nvidia``, this defaults to
-``sm_60``. For ``CHPL_GPU=amd`` it needs to be set explicitly. The environment
-setting can be overridden by the ``--gpu-arch`` compiler flag.  For a list of
-possible values please refer to "processor" values in `this table in the LLVM
-documentation <https://llvm.org/docs/AMDGPUUsage.html#processors>`_ for AMD or
-the `CUDA Programming Guide
+If the type of GPU is not detected you may set the ``CHPL_GPU`` environment
+variable manually to either ``nvidia`` or ``amd``.  ``CHPL_GPU`` may also
+manually be set to ``cpu`` to use `CPU as Device mode`_.
+
+Based on the value of ``CHPL_GPU``, Chapel's build system will also attempt to
+automatically detect the path to the relevant runtime. If it is not
+automatically detected (or you would like to use a different installation) you
+may set ``CHPL_CUDA_PATH`` and/or ``CHPL_ROCM_PATH`` explicitly.
+
+The CHPL_GPU_ARCH environment variable can be set to control the desired GPU
+architecture to compile for. The default value is ``sm_60`` for
+``CHPL_GPU_CODEGEN=cuda``. You may also use the ``--gpu-arch`` compiler flag to
+set GPU architecture. For a list of possible values please refer to "processor"
+values in `this table in the LLVM documentation
+<https://llvm.org/docs/AMDGPUUsage.html#processors>`_ for AMD or the `CUDA
+Programming Guide
 <https://docs.nvidia.com/cuda/cuda-c-programming-guide/#features-and-technical-specifications>`_
 for NVIDIA.
 
-``CHPL_RT_NUM_GPUS_PER_LOCALE`` can be used to control the number of GPU
-sublocales created. For example, in a system where 8 physical GPUs per node,
-setting this environment variable to 4 will make Chapel use only the first 4
-GPUs. Setting this number to a value higher than the number of GPUs per node
-results in an error; except for the ``CHPL_GPU==cpu`` mode, where this variable
-can be set to any non-negative number (See :ref:`below <cpu-mode-label>` for
-more information on this mode).
 
-GPU Support Features
---------------------
+CPU as Device Mode
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The ``CHPL_GPU`` environment variable can be set to ``cpu`` to enable many GPU
+features to be used without requiring any GPUs and/or vendor SDKs to be
+installed. This mode is mainly for initial development steps or quick feature
+tests where access to GPUs may be limited. In this mode:
+
+* the compiler will generate GPU kernels from order-independent loops normally,
```

Comment:
No other subsequent bullets have commas. Generally (nitpick), all the bullets here have inconsistent punctuation.

Reply:
This is a section Engin wrote that I moved, but I agree we should try for a consistent style within the list and more broadly within the document.

Turning a bulleted list into a kind of long sentence with things comma'd together is something I've done/seen in the past on academic papers and the like.  So I'm not really sure what's considered the best practice stylistically.

But anyway, for now I'm going to turn each bullet into a separate sentence given that's what we do elsewhere.